repository/src/main/java/org/apache/atlas/repository/store/graph/v2/preprocessor/contract/DataContract.java

```diff
@@ -0,0 +1,228 @@
+package org.apache.atlas.repository.store.graph.v2.preprocessor.contract;
+
+import java.lang.String;
+import java.util.*;
+import java.util.regex.Matcher;
+import java.util.regex.Pattern;
+
+import com.fasterxml.jackson.annotation.*;
+import com.fasterxml.jackson.core.JsonProcessingException;
+import com.fasterxml.jackson.databind.MapperFeature;
+import com.fasterxml.jackson.databind.ObjectMapper;
+import org.apache.atlas.AtlasErrorCode;
+import org.apache.atlas.exception.AtlasBaseException;
+import org.apache.commons.lang.StringUtils;
+
+import javax.validation.*;
+import javax.validation.constraints.NotNull;
+import java.util.Set;
+
+import static org.apache.atlas.AtlasErrorCode.*;
+
+
+@JsonIgnoreProperties(ignoreUnknown = true)
+@JsonPropertyOrder({"kind", "status", "template_version", "data_source", "dataset", "type", "description", "owners",
+        "tags", "certificate", "columns"})
+public class DataContract {
+    @Valid @NotNull
+    public String                               kind;
+
+    public STATUS                               status;
+
+    @JsonProperty(value = "template_version", defaultValue = "0.0.1")
+    public String                               templateVersion;
+    @Valid @NotNull
+    public String                              data_source;
+    @Valid @NotNull
+    public String                              dataset;
+    @Valid @NotNull
+    public DATASET_TYPE                        type;
+    public String                              description;
+    public List<String>                        owners;
+    public List<BusinessTag>                   tags;
+    public String                              certificate;
+    @Valid
+    public List<Column>                        columns;
+    private Map<String, Object>                unknownFields = new HashMap<>();
+
```

Comment:
can we mark these fields as private

Reply:
I have marked these field as private and added setter for all these attributes as the Jackson library will deserialize the json.